src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AspNet/AsyncControllerActionInvoker_BeginInvokeAction_Integration.cs

```diff
@@ -56,8 +58,15 @@ public class AsyncControllerActionInvoker_BeginInvokeAction_Integration
             {
                 if (HttpContext.Current != null)
                 {
-                    scope = AspNetMvcIntegration.CreateScope(controllerContext.DuckCast<ControllerContextStruct>());
+                    var duckedControllerContext = controllerContext.DuckCast<ControllerContextStruct>();
+                    HttpContext.Current.Request.PrepareArgsForWaf(duckedControllerContext.RouteData);
```

Comment:
It doesn't look like you're using the return value? What is the purpose of this call?

Reply:
yes thank you!